Microsoft.NET.Build.Containers/ContainerHelpers.cs

```diff
@@ -141,4 +141,64 @@ public static bool NormalizeImageName(string containerImageName, [NotNullWhen(fa
             return false;
         }
     }
+
+    [Flags]
+    public enum ParsePortError
+    {
+        MissingPortNumber,
+        InvalidPortNumber,
+        InvalidPortType,
+        UnknownPortFormat
+    }
+
+    public record ParsePortResult(bool success, Port? port, ParsePortError? parseErrors);
+
+    public static ParsePortResult ParsePort(string? portNumber, string? portType)
+    {
+        ParsePortError? errors = null;
+        int port = 0;
+        if (String.IsNullOrEmpty(portNumber))
+        {
+            errors = ParsePortError.MissingPortNumber;
+        }
+        else if (!int.TryParse(portNumber, out port))
+        {
+            errors = ParsePortError.InvalidPortNumber;
+        }
+
+        if (!Enum.TryParse<PortType>(portType, out PortType t))
```

Comment:
Is the spec case sensitive?

```suggestion
        if (!Enum.TryParse(portType, ignoreCase: true, out PortType t))
```
?

Reply:
it is, I've noted so on the enum